src/basics.md

```diff
@@ -277,6 +278,54 @@ fn main() {
 }
 ```
 
+[ex-rand-passwd]: #ex-rand-passwd
+<a name="ex-rand-passwd"></a>
+## Create random passwords from a set of allowed characters
+
+[![rand-badge]][rand]
+
+Randomly generates a string of given length ASCII characters in the range `A-Z, a-z, 0-9`, with [`gen_ascii_chars`] or [`choose`].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+# 
+# error_chain! {
+#     foreign_links {
+#         String(std::string::FromUtf8Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let s: String = thread_rng().gen_ascii_chars().take(30).collect();
+    println!("{}", s);
+
+    let alphanum_bytestring =
```

Comment:
I'd suggest implementing an iterator type like the [`AsciiGenerator`](https://docs.rs/rand/0.3.18/src/rand/lib.rs.html#670-684) instead that would take an rng in a constructor and for added merrit a "charset" (a hardcoded bytestr for our purposes without anych dynamic generation with ranges and chain) as an argument. 

Then it would be used almost identically as the `gen_ascii_chars`